sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -427,17 +427,19 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
       pluginContext: PluginContext,
       extraConf: java.util.Map[String, String]): Unit = {
     try {
-      if (Cuda.getComputeCapabilityMajor < 6) {
-        throw new RuntimeException(s"GPU compute capability ${Cuda.getComputeCapabilityMajor}" +
-          " is unsupported, requires 6.0+")
-      }
       // if configured, re-register checking leaks hook.
       reRegisterCheckLeakHook()
 
       val sparkConf = pluginContext.conf()
       val numCores = RapidsPluginUtils.estimateCoresOnExec(sparkConf)
       val conf = new RapidsConf(extraConf.asScala.toMap)
 
+      // Fail if the device's compute capability is less than minimum supported
+      if (Cuda.getComputeCapabilityMajor < conf.minCudaComputeCapabilitySupported) {
+        throw new RuntimeException(s"GPU compute capability ${Cuda.getComputeCapabilityMajor}" +
+          s" is unsupported, requires ${conf.minCudaComputeCapabilitySupported}+")
```

Comment:
I wonder if we can get it of our native build somehow instead of a config.

Reply:
I agree with Gera and to me this approach seems good but I speak from my limited understanding.

Reply:
@parthosa @mythrocks my remark is not regarding the device part of the "equation" / comparison [getComputeCapabilityMajor](https://github.com/rapidsai/cudf/blob/d4368e98a4b92ade651a5f5df98035a297658f16/java/src/main/native/src/CudaJni.cpp#L202-L213) is already querying the current device

My improvement suggestion is to avoid introducing an artificial config for the supported part of the comparison. It may technically be even inaccurate to reduce it to a single comparison point when GPU_ARCHS colon-separated list had a gap in major versions. It will work fine with official builds.  

As part of the build in spark-rapids-jni we could run:
```
~/gits/NVIDIA/spark-rapids-jni/target$ cuobjdump ./libcudf-install/lib64/libcudf.a | grep 'arch =' | sort -u
arch = sm_86
~/gits/NVIDIA/spark-rapids-jni/target$ cuobjdump ./libcudfjni/libcudfjni.a | grep 'arch =' | sort -u
arch = sm_86
```
There could be better ways. The bottom line: the list of sm_XY can be made part of spark-rapids-jni-version-info.properties
/ cudf-java-version-info.properties for easy  querying in the Plugin.

Reply:
Hmm.  I do see the merit in what @gerashegalov is proposing.  (One trivial snag is there is that there will likely be multiple `arch=` lines for a released `libcudf` version. We'd have to sort/cut to determine the min version.)

@parthosa and I were trying to assign the responsibility of determining the minimum device version to the place where the limitation originates.  It's not the Plugin that decides what's incompatible, it's `libcudf`.  IMHO, the CUDF-JNI code shouldn't "start up" on unsupported platforms, for fear of bad results.  Putting this in `getDevice`/`setDevice` seemed easiest.

Maybe one approach needn't preclude the other.  Having this information also available in `version-info.properties` makes this query-able in the field.

Reply:
@mythrocks Pushing the assert logic down to cuDF Java and / or spark-rapids-jni  makes sense to me. My focus is to avoid making up configs instead of getting this information from the buildtime or the runtime.

Reply:
> Having this information also available in version-info.properties

+1 for this, that could be very helpful to verify whether or not a particular GPU arch has been included in the build or not.

Reply:
Demo for a two-arch build:
```bash
~/gits/NVIDIA/spark-rapids-jni$ ./build/build-in-docker clean install -DCPP_PARALLEL_LEVEL=6 -DGPU_ARCHS='86;75'  -Dlibcudf.clean.skip=false -DskipTests
~/gits/NVIDIA/spark-rapids-jni$ find target -name libcudfjni.a -o -name libcudf.a | DOCKER_RUN_EXTRA_ARGS=-i ./build/run-in-docker xargs -t -n 1 bash -c 'cuobjdump $1 || exit 0' _ | grep 'arch = ' | awk -F_ '{print $2}' | sort -n -u
bash -c cuobjdump $1 || exit 0 _ target/libcudfjni/libcudfjni.a 
bash -c cuobjdump $1 || exit 0 _ target/libcudf-install/lib64/libcudf.a 
75
86
```